src/Pages/ProductEdit.tsx

```diff
@@ -0,0 +1,274 @@
+              display: "flex",
+              justifyContent: "center",
+              alignItems: "center",
+              flexDirection: "column",
+            }}
+          >
+            <Typography variant="h4" component="div" gutterBottom>
+              Edit Product
+            </Typography>
+
+            <TextField
+              sx={{ margin: "20px", width: "90%" }}
+              label="Producent name"
+              {...register("producentName")}
+              error={!!errors.producentName}
+              helperText={errors.producentName?.message}
+            />
+
+            <TextField
+              sx={{ margin: "20px", width: "90%" }}
+              label="Strain name"
+              {...register("strainName")}
+              error={!!errors.strainName}
+              helperText={errors.strainName?.message}
+            />
+
+            <TextField
+              sx={{ margin: "20px", width: "90%" }}
+              select
+              label="Genetics"
+              {...register("genetics")}
+              error={!!errors.genetics}
+              helperText={errors.genetics?.message}
+            >
+              {geneticsOptions.map((option) => (
+                <MenuItem key={option.value} value={option.value}>
+                  {option.label}
+                </MenuItem>
+              ))}
+            </TextField>
+
+            <TextField
+              sx={{ margin: "20px", width: "90%" }}
+              label="THC"
+              {...register("thcLevel")}
+              error={!!errors.thcLevel}
+              helperText={errors.thcLevel?.message}
+            />
+
+            <TextField
+              sx={{ margin: "20px", width: "90%" }}
+              label="CBD"
+              {...register("cbdLevel")}
+              error={!!errors.cbdLevel}
+              helperText={errors.cbdLevel?.message}
+            />
+
+            <TextField
+              sx={{ margin: "20px", width: "90%" }}
+              select
+              fullWidth
+              label="Terpen"
+              {...register("terpen")}
+              error={!!errors.terpen}
+              helperText={errors.terpen?.message}
+            >
+              {terpenOptions.map((option) => (
+                <MenuItem key={option.value} value={option.value}>
+                  {option.label}
+                </MenuItem>
+              ))}
+            </TextField>
+
+            <TextField
+              sx={{ margin: "20px", width: "90%" }}
+              fullWidth
+              label="Description"
+              {...register("description")}
+              error={!!errors.description}
+              helperText={errors.description?.message}
+            />
+
+            <Divider sx={{ borderColor: "black", borderStyle: "solid" }} />
+            <Box display="flex" gap={2}>
+              <Button
+                type="submit"
+                variant="contained"
+                color="success"
+                onClick={() => navigate(`/`)}
```

Comment:
? masz to samo w onSubmit, teraz masz konflikt, prawdopodbnie szybciej Cie przekierowuje niz wyslesz dane i sprawdzisz czy poprawnie sie wyslaly
```suggestion
```